squidDrone/tests/i2c/i2c_interface_tests.cpp

```diff
@@ -0,0 +1,139 @@
+#include "gtest/gtest.h"
+#include "gmock/gmock.h"
+#include "i2c.hpp"
+
+namespace{
+
+  class I2CTests: public ::testing::Test{
+  protected:
+    virtual void SetUp(){
+      unit_under_test_ = std::make_unique<i2c::I2C>();
+    }
+
+    std::unique_ptr<i2c::I2C> unit_under_test_;
+
+    uint8_t address = 0x10;
+    uint16_t byte_size = 3;
+    uint32_t timeout = 2;
+    std::vector<uint8_t> data = {1, 2};
+
+    i2c::I2CStatus result_status;
+    std::vector<uint8_t> result_data;
+  };
+
```

Comment:
read is not tested with illegal equivalency class . timeout = 0

Reply:
It is already tested. See here: https://github.com/elheck/Squiddrone/blob/e578c51813e066b04af12f4a910e87de8dfaf400/squidDrone/tests/i2c/i2c_tests.cpp#L66

Reply:
My bad